web/src/app/queries/queries.js

```diff
@@ -3,12 +3,37 @@ import { Expire } from 'avenger/lib/cache/strategies';
 import t from 'tcomb';
 import API from 'API';
 
+const defaultExchanges = ['Bitfinex', 'Bitstamp', 'Kraken'];
+
+const checkExchanges =
+  (exchanges) =>
+  (exchanges.length === 1 && exchanges[0] === '') ?
+  JSON.stringify(defaultExchanges) : JSON.stringify(exchanges);
```

Comment:
I'm not sure I get this. Is it because of the `Dropdown.onChange` calling back with `['']` as value when no exchange is selected? I'd prefer to see this handled at the component level, so that this special treatment doesn't pollute the rest of the app